src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -111,6 +115,17 @@ private void updateValue(
     } else {
       this.parameterMap.put(key, defaultValue);
     }
+
+    if (enforceDefault) {
```

Comment:
Should we put this at the beginning of the function?

Reply:
Make sense, resolved.

Reply:
unclear why this should be at the beginning of the function? You are missing out on validation of any override passed in via parameterOverrides or via props.

Reply:
Ah yes, reverted to original order.